src/GitLink/Providers/VisualStudioTeamServicesProvider.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+using GitTools.Git;
+
+namespace GitLink.Providers
+{
+    public class VisualStudioTeamServicesProvider : ProviderBase
+    {
+        private readonly Regex _visualStudioTeamServicesRegex = new Regex(@"(?<url>(?<companyurl>(?:https://)?(?<accountname>([a-zA-Z0-9\-\.]*)?)\.visualstudio\.com/)(?<project>[a-zA-Z0-9\-\.]*)/?_git/(?<repo>[^/]+))");
+
+        public VisualStudioTeamServicesProvider() 
+            : base(new GitPreparer())
+        {
+        }
+
+        public override string RawGitUrl
+        {
+            get
+            {
+                return "";
+            }
+        }
+
+        public override bool Initialize(string url)
+        {
+            var match = _visualStudioTeamServicesRegex.Match(url);
+
+            if (!match.Success)
+            {
+                return false;
+            }
+
+            CompanyName = match.Groups["accountname"].Value;
+            CompanyUrl = match.Groups["companyurl"].Value + "DefaultCollection/";
```

Comment:
Thanks. If this is verified, then it's all good. Or it's up to the next person to create a separate PR for this ;-)